File: app/src/main/java/com/mju/csmoa/SignInActivity.java
package com.mju.csmoa;

import android.content.Intent;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.mju.csmoa.databinding.ActivitySignInBinding;

import java.util.Objects;

public class SignInActivity extends AppCompatActivity {

    private ActivitySignInBinding binding;

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySignInBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        init();

//        getWindow().setStatusBarColor(Color.TRANSPARENT); // statusBar transparent
//        getWindow().setNavigationBarColor(Color.TRANSPARENT); // navigationBar transparent
//        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
//                | View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
//                | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
    }

    private void init() {

        // email input
        binding.textInputEditTextSignInEmailInput.addTextChangedListener(new TextWatcher() {
            @Override // 입력하기 전에
            public void beforeTextChanged(CharSequence s, int start, int count, int after) { }

            @Override // 입력란에 변화가 있을 시
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
                // email pattern matched
                if (Objects.requireNonNull(binding.textInputEditTextSignInEmailInput.getText()).toString().trim().matches(emailPattern)) {
//                    binding.textInputEditTextSignInEmailInput.setBackgroundTintList(ColorStateList.valueOf(Color.BLUE));
                    binding.textInputLayoutSignInEmailInputLayout.setError(null);
                    binding.textInputEditTextSignInEmailInput.setCompoundDrawablesWithIntrinsicBounds(
                            null, null, ContextCompat.getDrawable(SignInActivity.this, R.drawable.ic_all_checked), null);
                } else { // email pattern not matched
                    binding.textInputLayoutSignInEmailInputLayout.setError("올바른 이메일 양식이 아닙니다");
                    binding.textInputEditTextSignInEmailInput.setCompoundDrawablesWithIntrinsicBounds(
                            null, null, null, null);
                }
            }

            @Override // 입력이 끝났을 때
            public void afterTextChanged(Editable s) { }
        });

         // signUp
        binding.buttonSignInSignUp.setOnClickListener((View view) -> {
            startActivity(new Intent(SignInActivity.this, SignUpActivity.class));
        });

        // 로그인 성공 -> 홈
        // TODO

        // 카카오 로그인
        // TODO

        // 구글 로그인
        // TODO
    }
}

File: app/src/main/java/com/mju/csmoa/SignInActivity.kt
package com.mju.csmoa;

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.mju.csmoa.databinding.ActivitySignInBinding

class SignInActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySignInBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySignInBinding.inflate(layoutInflater)
        setContentView(binding.root)
        init()

//        window.statusBarColor = Color.TRANSPARENT
//        window.navigationBarColor = Color.TRANSPARENT
//        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
//        | View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
//        | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
    }

    private fun init() {

        // Set TextWatcher
        val textWatcher = object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) { }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                val emailPattern = Regex("[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+")
                // check if email pattern mathed
                if (binding.textInputEditTextSignInEmailInput.text.toString().trim().matches(emailPattern)) {
                    binding.textInputLayoutSignInEmailInputLayout.error = null
                    binding.textInputEditTextSignInEmailInput.setCompoundDrawablesWithIntrinsicBounds(
                        null, null, ContextCompat.getDrawable(baseContext, R.drawable.ic_all_checked), null);
                } else {


                    binding.textInputLayoutSignInEmailInputLayout.error = "올바른 이메일 양식이 아닙니다"
                    binding.textInputEditTextSignInEmailInput.setCompoundDrawablesWithIntrinsicBounds(
                        null, null, null, null);
                }
            }

            override fun afterTextChanged(s: Editable?) { }
        }

        // email input
        binding.textInputEditTextSignInEmailInput.addTextChangedListener(textWatcher)

        // signUp
        binding.buttonSignInSignUp.setOnClickListener {
            startActivity(Intent(this, SignUpActivity::class.java))
        }

        // 로그인 성공 -> 홈
        binding.buttonSignInSignIn.setOnClickListener {
            startActivity(Intent(this, HomeActivity::class.java))
            finish()
        }

        // 카카오 로그인
        // TODO

        // 구글 로그인
        // TODO
    }
}